pilot/pkg/bootstrap/server.go

```diff
@@ -827,6 +827,9 @@ func (s *Server) cachesSynced() bool {
 	if !s.configController.HasSynced() {
 		return false
 	}
+	if !s.workloadTrustBundle.HasSynced() {
```

Comment:
My understand of how this works (and correct me if I am wrong..): Istiod starts up, we try to fetch the bundle. If it succeeds we are good to go. If it fails, we wait _30 minutes_ to try again. After 30s we mark ourselves ready.

So I am not sure what the point of waiting 30s to mark ourselves ready. We should either aggressively retry or just mark ourselves ready immediately (since we are not retrying in the meantime anyways)?

Reply:
I understand. My vote would be to mark ourselves ready immediately. Note that we poll each endpoint configured for 10 whole seconds waiting for it to become available. Lack of availability after 10s usually implies (a) Network Issues (congestion/connectivity) (b) Endpoint is not up. In these scenarios, I think it is better to not be aggressive and back off, and try again after 30 minutes.

Reply:
So if the spiffe bundle is down for ~10 seconds (possible with 5 nines reliability), I will fail to fetch it. We mark ourselves ready. New proxies connect, get no root cert, and all mtls will fail as a result with obscure tls errors. We will not recover for 30 minutes.

This seems like a really dangerous idea.. am I missing some reason that this is safe?

It seems equivalent to failing open if we fail to read AuthorizationPolicy from k8s api-server

Reply:
Hmm..I see the concern here. This isn't safe in the scenario you described..however, as we discussed earlier, we don't want to unnecessarily delay Istiod from starting up for this. Perhaps we could have an exponential backoff here in event of failure? Until we get all certs from all remote Trust Anchors?